### PR TITLE
DM-31384: Add --log-label option

### PR DESCRIPTION
**Pull request by timj**
## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/daf/butler/cli/cliLog.py [hunk 1]
```diff
@@ -84,7 +85,8 @@ class CliLog:
     that need to be closed on reset."""
 
     @classmethod
-    def initLog(cls, longlog: bool, log_tty: bool = True, log_file=()):
+    def initLog(cls, longlog: bool, log_tty: bool = True, log_file: Tuple[str, ...] = (),
+                log_label: Optional[Dict] = None):
```
Comment (andy-slac) [inline python/lsst/daf/butler/cli/cliLog.py]: Maybe `Dict[str, str]`?

File: python/lsst/daf/butler/cli/opt/options.py [hunk 2]
```diff
@@ -122,6 +122,13 @@ def makeCollectionTypes(context, param, value):
                                     " JSON log records will be written, else formatted text log records"
                                     " will be written. This file can exist and records will be appended.")
 
+log_label_option = MWOptionDecorator("--log-label",
+                                     default=None,
+                                     multiple=True,
+                                     callback=split_kv,
+                                     type=str,
+                                     help="Keyword=value pairs to add to JSON log records.")
```
Comment (andy-slac) [inline python/lsst/daf/butler/cli/opt/options.py]: I think it's not only about JSON, should we say "add to MDC of log records"?

**Review by andy-slac — APPROVED** [review]
Looks great, couple of minor comments.

File: python/lsst/daf/butler/cli/opt/options.py (hunk 2 above)
Comment (timj) [reply python/lsst/daf/butler/cli/opt/options.py]: Okay. We don't yet allow people to specify the template for long-log but in the future we could do so and that would make the content of these labels available to the formatted logs.

**Pull request merged**